contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol

```diff
@@ -258,7 +258,7 @@ contract OCR2DRRegistry is ConfirmedOwner, OCR2DRRegistryInterface, ERC677Receiv
     }
     uint256 executionGas = s_config.gasOverhead + s_config.gasAfterPaymentCalculation + billing.gasLimit;
     // (1e18 juels/link) (wei/gas * gas) / (wei/link) = juels
-    uint256 paymentNoFee = (1e18 * tx.gasprice * executionGas) / uint256(weiPerUnitLink);
+    uint256 paymentNoFee = (1e18 * billing.gasPrice * executionGas) / uint256(weiPerUnitLink);
```

Comment:
Just double check for possible overflow errors. I know you probably reused the code from VRF, but maybe it makes sense to use `SafeMath` library.

Reply:
[Solidity v0.8](https://docs.soliditylang.org/en/v0.8.15/080-breaking-changes.html) actually handles arithmetic overflows and underflows natively :slightly_smiling_face: 
![image](https://user-images.githubusercontent.com/28818476/205398837-c93db602-4214-4948-b816-275f6d380623.png)